crnk-gen/crnk-gen-openapi/src/test/java/io/crnk/gen/openapi/OpenAPIGeneratorTestBase.java

```diff
@@ -72,9 +72,10 @@ void compare(String expectedSourceFileName, String actualSourcePath) throws IOEx
   }
 
   private static void assertOperationResponseCodes(Operation operation, List<String> codes) {
-  	if(operation == null){
-  		throw new IllegalStateException();
-	}
+    // Some operations will be null in cases where a JsonApiResource has postable, patchable,
+    // readable, deleteable set to false. In that case, do not perform any response checks
+    if(operation == null) { return; }
```

Comment:
Should the null operations even be getting this far?

Reply:
In order to avoid a bunch of `null` checks sprinkled everywhere else, I thought it best to skip here. The original assumption was that a `null` Operation would be invalid, when in fact there *should* be null operations when that operation isn't permitted.

![image](https://user-images.githubusercontent.com/7207744/69268492-9e34ef00-0ba5-11ea-9d08-2c2bdae21dee.png)

Reply:
I see

Reply:
Should the paths even exist?

Reply:
Paths? Yes, one or more of the individual Operations on the path might not though. 

Example: something that is immutable in a system may allow `GET /thing/{ThingId}` and `DELETE /thing/{ThingId}` but not `PATCH /thing/{ThingId}`. In the screenshot above, we'd want to skip PATCH response validation because `getPatch()` returns null.